homeassistant/components/zeversolar/sensor.py

```diff
@@ -0,0 +1,100 @@
+"""Support for the JustNimbus platform."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, UnitOfEnergy, UnitOfPower
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .coordinator import ZeverSolarCoordinator
+from .entity import ZeverSolarEntity
+
+
+@dataclass
+class ZeverSolarEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[ZeverSolarCoordinator], Any]
+
+
+@dataclass
+class ZeverSolarEntityDescription(
+    SensorEntityDescription, ZeverSolarEntityDescriptionMixin
+):
+    """Describes ZeverSolar sensor entity."""
+
+
+SENSOR_TYPES = (
+    ZeverSolarEntityDescription(
+        key="pac",
+        name="Power in watts",
+        icon="mdi:solar-power-variant",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.POWER,
+        value_fn=lambda coordinator: coordinator.data.pac,
+    ),
+    ZeverSolarEntityDescription(
+        key="energy_today",
+        name="Energy Today",
+        icon="mdi:home-battery",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
```

Comment:
If this daily resets to 0, it should be `total_increasing`. If not, this would probably need an attribute to indicate its reset time.

```suggestion
        state_class=SensorStateClass.TOTAL_INCREASING,
```